### ISSUE TITLE
periodic interaction 1D materials 

### ISSUE THREAD

**Issue by simonganne01** [pyscf/pyscf#2224]
Hello everyone, 

I am simulating polyyne (a 1D carbon chain with alternating single and triple bonds). I want to simulate 1D but I noticed that 1D simulations are not yet implemented for DFT. In order to avoid this I just made the unit cell big in the two directions perpendicular to the chain. However when I calculate the Coulomb repulsion of the wannier orbitals (calculated with wannier90 and pywannier90 as implemented in pyscf) I get rising values when increasing the number of kpoints along the x-axis. (The number of kpoints along the y- and 
![image](https://github.com/pyscf/pyscf/assets/93279056/7ec86040-fb17-4b06-9f8c-0669cce9f2e4)
z-axis are fixed at 1 because we do not want to take into account the periodic effects along these directions anyway.)
![image](https://github.com/pyscf/pyscf/assets/93279056/7ec86040-fb17-4b06-9f8c-0669cce9f2e4)

The slope of this increase seems to decrease as the unit cell grows bigger, so my hypotheses is that it has something to do with the interaction with neighbouring chains. Anyone else an explanation for this phenomenon?

**Comment by tberkel** [comment]
Leaving the Wannier orbital repulsion aside first, does the total energy
behave reasonably with the number of kpoints and the box size?

**Comment by simonganne01** [comment]
Yes, the total energy seems to converge just fine (different colors are different basis sets: DZ, TZ and QZ): 

![image](https://github.com/pyscf/pyscf/assets/93279056/d1eaa692-bd27-4d2c-beba-e3e2454f6f46)

**Comment by simonganne01** [comment]
yes that was my first guess to, but then the order of magnitude will be all wrong. Furthermore, when I use more k-points in the perpendicular direction, the linear increase is less prominent. thus, normalisation wrt k-points wil not solve the problem. 

The Nyquist theorem explains why the Wannier orbitals would have less periodic interaction when using more k-points in the perpendicular direction, but I still don't have an explanation for the linear increase when using 1 k-point in the perpendicular directions.

**Comment by simonganne01** [comment]
I do not have this effect using VASP btw, so it must have to do with the GTO's or the density fitting or something else PySCF specific.

**Comment by tberkel** [comment]
Hm, maybe it has something to do with the neutralizing background and the treatment of G=0 in the Coulomb interaction? One debugging idea: instead of increasing the number of k-points, what happens if you make the equivalent supercells in the x direction? Do you get the same results?

**Comment by sunqm** [comment]
https://github.com/pyscf/pyscf/blob/a64f740c978eb6cc10eec53f0c50098fc4f543b2/pyscf/pbc/scf/khf.py#L248 This is how coulomb energy computed in pyscf. Do you compute the Coulomb energy using the API `get_j` or your custom script?

**Comment by verena-neufeld** [comment]
Hi @simonganne01 , thanks for investigating and reporting this. Please let us know if you have an update on this.

**Comment by simonganne01** [comment]
> Hi @simonganne01 , thanks for investigating and reporting this. Please let us know if you have an update on this.
I will certainly investigate this further, currently I'm busy with my last exams.

**Comment by simonganne01** [comment]
Is it a possibility that this error is caused by a reason related to section 2.5 (Coulomb divergence correction) of the following paper: https://doi.org/10.1021/acs.jctc.0c00704 ? since this effect is strongly decreased when I take into account more kpoints in the directions perpendicular to my 1D chain.

**Comment by simonganne01** [comment]
Ok, I think that I indeed need to include the Heads interactions term. But I am wondering why this equation in the GW code doest include an extra omega (cell volume) x Nk. As described in the paper on G0W0. 

`Del_00 = 2./np.pi * (6.*np.pi**2/gw.mol.vol/nkpts)**(1./3.) * (eps_inv_00 - 1.)`

**Comment by simonganne01** [comment]
Furthermore, could it be that my problems would be solved if I adapted the softwae to include mixed density fitting? I think so but I don't want to waste my time if it is not the case.

**Comment by tberkel** [comment]
You shouldn't have to add anything yourself. The periodic integrals in PySCF are calculated using the Ewald interaction, which, in Fourier space, is equivalent to neglecting the G=0 component of the Coulomb interaction. Equivalently, this contribution is neglected by removing the net charge of the AO pair densities, but this is done in the code. The type of DF you use (FFTDF, GDF, or MDF) doesn't change this defining feature of the integrals; it only speeds up the calculation depending on your basis set, pseudopotential, etc.

**Comment by simonganne01** [comment]
I am now trying to implement the MDF module to calculate the screened coulomb interaction to see if this solves my problem. I noticed that the number of auxiliary basis functions is equal to naux = "number of auxiliary gausian basis functions"+"number of plane waves"*2. Where does the multiplication by two come from?
Additionally: What is the difference between the pw_loop function and the ft_loop? 
Thanks for the help!

**Comment by simonganne01** [comment]
Ok, new chapter in the search for converged coulomb interactions in 1D. As mentioned before, I use pywannier90 to calculate maximally localized wannier orbitals of materials like polyacetelyne (a kind of 1D carbon chain). After that I calculate the coulomb and screened coulomb interactions for those wannier orbitals. this is all done using the gaussian density fitting technique. I started truly 1D simulations (cell.dimension = 1) in order to try and eliminate the spurious periodic interactions which I believe to cause the linear increase in Coulomb interaction as mentioned in my first post here. However, I noticed that the convergence in terms of kpoints is insanely slow (see figure: V for bare coulomb, W for screened Coulomb). If I check if the total energy is converged (figure total energy), this convergence is a lot faster. Does anyone have insight into why this might be and how I can improve the convergence of the coulomb interactions? Thanks in advance.
 [V_0_0_0_0.pdf](https://github.com/user-attachments/files/17524433/V_0_0_0_0.pdf)
[W_0_0_0_0.pdf](https://github.com/user-attachments/files/17524434/W_0_0_0_0.pdf)
[energy.pdf](https://github.com/user-attachments/files/17524435/energy.pdf)

**Comment by simonganne01** [comment]
I kind of figured out that the problem is related to the contribution at q=0. This contribution seems to large. In the plot below I put the contribution t the coulomb repulsion in function of q (difference in kpoints). I id this for different kpoints grids. You can see the sampling becomes denser which will result in a more accurate integral under the curve. But it also becomes apparent that the contribution at q=0 results in a large overestimation of the area under the curve. I get a lot faster convergence by putting this integral to zero. In the limit of infinite kpoints this should result in the correct integral even if we neglect the contribution at q=0. so I think I converge to a correct value But has anyone any insight into why this part seems to be wrong at q=0? 
[V(q)_0_0_0_0_no_log.pdf](https://github.com/user-attachments/files/17993840/V.q._0_0_0_0_no_log.pdf)

**Comment by simonganne01** [comment]
I think I need to add a prefactor to the trms at q=0 because of the divergence and the special threatment there but I do not know what the prefactor should be. Furthermore, I think this prefactor should be integrated in the code.

**Comment by jeanwsr** [comment]
Hi @simonganne01 , has this issue been resolved?

**Comment by simonganne01** [comment]
No, I am no longer working with PySCF. So I will not be able to resolve it.
